app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java

```diff
@@ -139,8 +141,8 @@ public Task<OrderedHashSet<String>> fetchFoundPins() {
         if (auth.getUid() == null) {
             throw new IllegalStateException("User must be logged in to fetch pins");
         }
-        return db.collection("users").document(auth.getUid()).collection("found").get()
-                .continueWithTask(task -> {
+        return db.collection("users").document(auth.getUid()).collection("found")
+                .orderBy("timestamp").get().continueWithTask(task -> {
```

Comment:
just for clarification, if we order by timestamp on database calls, why do we need to insert into the `OrderedHashSet`?

Reply:
since we need to ensure it maintains it's order while preserving the O(1) access time of a HashSet. A plain HashSet won't preserve an order and a LinkedHashSet takes O(n) time to access an element at a given index.